client/lib/media/store.js

```diff
@@ -31,6 +31,15 @@ function receiveSingle( siteId, item, itemId ) {
 		}
 
 		MediaStore._pointers[ siteId ][ itemId ] = item.ID;
+
+		const previousMediaItem = MediaStore._media[ siteId ][ itemId ];
```

Comment:
Hmm, looking into this more, `itemId` in `receiveSingle` can be something different than just the id of a transient media (e.g. [MediaActions.fetch](https://github.com/Automattic/wp-calypso/blob/master/client/lib/media/actions.js#L40-L40) method).

In ccd823b, I just renamed `previousMediaItem` to `maybeTransientMediaItem` as media item with the key `transient` is always transient. If the media object doesn't contain this key, don't modify anything.

What do you think?